app/Http/Controllers/SignLocallyController.php

```diff
@@ -235,8 +241,25 @@ public function downloadSignedFile(Request $request)
         }
 
         $data = $this->eidEasyApi->downloadSignedFile($docId);
+        $fileName = $data['filename'];
+
+        $signedFilesContent = [];
+        foreach ($data['signed_files'] as $file) {
```

Comment:
I see there is a chance that $data['signed_files'] can be undefined key on $data array in case of error coming from API request. Let's add check for that key to exists before running into foreach to prevent critical error.

Reply:
fixed